src/pm-conversations/recentPmConversationsReducer.js

```diff
@@ -1,17 +1,42 @@
 /* @flow strict-local */
+import isEqual from 'lodash.isequal';
+
 import type { Action, RecentPrivateConversationsState } from '../types';
-import { REALM_INIT } from '../actionConstants';
+import { REALM_INIT, EVENT_NEW_MESSAGE } from '../actionConstants';
 import { NULL_ARRAY } from '../nullObjects';
 
 const initialState: RecentPrivateConversationsState = NULL_ARRAY;
 
+const newMessage = (state, action) => {
+  if (action.message.type !== 'private') {
+    return state;
+  }
+
+  const userIds = action.message.display_recipient.map(recipient => recipient.id);
+  const index = state.findIndex(item => isEqual(item.user_ids, userIds));
+  const oldMaxMsgId = index < 0 ? null : state[index].max_message_id;
+  const item = {
+    user_ids: userIds,
+    max_message_id: Math.max(action.message.id, oldMaxMsgId ?? -Infinity),
```

Comment:
Hmm, have we thought through storing `-Infinity` in the Redux state; in particular, what happens when it's saved and then loaded from storage? It looks like it doesn't round-trip through JSON:

```
JSON.parse(JSON.stringify({ a: -Infinity }))
// {a: null}
```